### PR TITLE
Abstracted container object using a new interface in worker

### PR DESCRIPTION
**Pull request by ritwik-g**
## What

- Added a new interface for container object.

## Why

- This interface allows us to represent only the actions required for us. And this will help us disassociate container run time specific implementation from main code. 

## How

- Currently an iterable logs and a method to make sure container cleanup happens are our requirements. So we have created an interface for this. We will create implementation of this depending up on container run time along with it's client interface implementation.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- It shouldn't since we are only doing refactoring

## Database Migrations

- None

## Env Config

- None

## Relevant Docs

- None

## Related Issues or PRs

- None

## Dependencies Versions

- None

## Notes on Testing

- Tested locally running a workflow.

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

**Review by vishnuszipstack — APPROVED** [review]

File: worker/src/unstract/worker/worker.py [hunk 1]
```diff
@@ -213,7 +215,8 @@ def run_container(
         except Exception as e:
             self.logger.error(f"Failed to run docker container: {e}")
             result = {"type": "RESULT", "result": None, "error": str(e)}
-        self._cleanup_container(container)
+        if container:
+            container.cleanup()
```
Comment (kirtimanmishrazipstack) [inline worker/src/unstract/worker/worker.py:219]: NIT: Is this clean up happeing always after particular image is run inside worker?

**Review by kirtimanmishrazipstack — APPROVED** [review]
LGTM

File: worker/src/unstract/worker/worker.py (hunk 1 above)
Comment (ritwik-g) [reply worker/src/unstract/worker/worker.py:219]: Yes.

**Pull request merged**